packages/libs/user-datasets/src/lib/Components/UploadForm.tsx

```diff
@@ -490,143 +492,144 @@ function UploadForm({
+                      value,
+                      'citation'
+                    );
+                    setPublications(updatedPublications);
+                  }}
+                />
+              );
+            })}
+            <OutlinedButton
+              text="Add Publication"
+              onPress={(event: React.MouseEvent<HTMLButtonElement>) => {
+                event.preventDefault();
+                setPublications((oldPublications) => [
+                  ...oldPublications,
+                  {} as UserDatasetPublication,
+                ]);
+              }}
+              icon={AddIcon}
+              styleOverrides={OutlinedButtonWDKStyle}
+            />
+          </div>
+          <div className="additionalDetailsFormSection additionalDetailsFormSection--data-set-hyperlinks">
+            <FieldLabel
+              htmlFor="data-set-publications-hyperlinks"
+              required={false}
+            >
+              Hyperlinks
+            </FieldLabel>
+            {hyperlinks.map((hyperlink, index) => {
+              const updateHyperlinksObject = createNestedInputUpdater({
+                nestedInputObject: hyperlinks,
+                index,
+              });
+              return (
+                <HyperlinkInput
+                  n={index}
+                  url={hyperlink.url}
+                  onAddUrl={(value: string) => {
+                    const updatedHyperlinks = updateHyperlinksObject(
+                      value,
+                      'url'
+                    );
+                    setHyperlinks(updatedHyperlinks);
+                  }}
+                  onRemoveHyperlink={(
+                    event: React.MouseEvent<HTMLButtonElement>
+                  ) => {
+                    event.preventDefault();
+                    const updatedHyperlinks = [...hyperlinks];
+                    updatedHyperlinks.splice(index, 1);
+                    setHyperlinks(updatedHyperlinks);
+                  }}
+                  text={hyperlink.text}
+                  onAddText={(value: string) => {
+                    const updatedHyperlinks = updateHyperlinksObject(
+                      value,
+                      'text'
+                    );
+                    setHyperlinks(updatedHyperlinks);
+                  }}
+                  description={hyperlinks[index]?.description}
+                  onAddDescription={(value: string) => {
+                    const updatedHyperlinks = updateHyperlinksObject(
+                      value,
+                      'description'
+                    );
+                    setHyperlinks(updatedHyperlinks);
+                  }}
+                  isPublication={hyperlinks[index]?.isPublication}
+                  onAddIsPublication={(value: boolean) => {
+                    const updatedHyperlinks = updateHyperlinksObject(
+                      value,
+                      'publication'
```

Comment:
```suggestion
                      'isPublication'
```

This was breaking the radio button.

Reply:
This would have been caught by TypeScript if we added `inputName: keyof T` to the factory. (This is the original.)

```
const createNestedInputUpdater = function <T>(
  props: AddNestedInputChildProps<T>
) {
  const { nestedInputObject, index } = props;

  return function (newValue: string | boolean, inputName: keyof T) {
    const updatedNestedInputObject = [...nestedInputObject];
    updatedNestedInputObject[index] = {
      ...nestedInputObject[index],
      [inputName]: newValue,
    };
    return updatedNestedInputObject;
  };
};
```